packages/tasit-action/src/contract/Action.js

```diff
@@ -108,6 +108,10 @@ export class Action extends Subscription {
     const baseEthersListener = async blockNumber => {
       try {
         const tx = await this.#tx;
+        if (!tx) {
+          console.log(`The action wasn't sent yet.`);
```

Comment:
How does this log statement in a dependency (our own) get handled with our removing log approach in the Tasit apps?

Reply:
Oh, I should have used the `warn` instead.
The console removal plugin used on `tasit` removes all `console.*` from code.
Not sure about the best way to deal with `warn`, not sure if make sense use `error` event for that. What do you think?

Reply:
Oh yes, that's a good point that this should be `warn`.

If this is something that would prevent the tx from going through, then passing this back to the caller as an error so they can retry the tx and update the UI accordingly probably makes sense.

Reply:
Oh, just thought again about it: It's happening when an action starts to be listening before send it (the `tx` field is assigned on the send function). I think that worths a test case, and maybe is a situation to be ignored?

Reply:
Yeah, that's worth a test case

I'm not sure if it's a situation to be ignored, though? Why would it be okay to ignore it?

Reply:
Should this gate merging?

Reply:
I've wrote this test case:

```javascript
    it("should be able to listen to an event before sending", async () => {
      const confirmationListener = sinon.fake(async message => {
        action.off("confirmation");
      });

      const errorListener = sinon.fake();

      action = sampleContract.setValue(rand);
      action.on("error", errorListener);
      action.on("confirmation", confirmationListener);

      await mineBlocks(provider, 2);

      await action.send();
      await action.waitForOneConfirmation();

      await mineBlocks(provider, 2);

      expect(confirmationListener.callCount).to.equal(1);
      expect(errorListener.called).to.be.false;
    });
```

That way, warn is being printed for each new block mined, I'm not sure if it's desirable. 
Since:
1) It's expected behavior;
2) I've improved the `signAndSend` function (that assigns the `this.#tx` field) to guarantee that it'll emit an error event if some error occurs;
Maybe we could remove this warn.

Reply:
> Should this gate merging?

I don't think so.

Reply:
Ahh ok - I forgot that the action not being sent yet was something we would want to happen sometimes.

Based on that, what I said earlier:

> If this is something that would prevent the tx from going through, then passing this back to the caller as an error so they can retry the tx and update the UI accordingly probably makes sense.

...didn't make sense. So I'm glad you wrote this test as a way of clarifying how we would run into this scenario. Thanks!

Reply:
> That way, warn is being printed for each new block mined, I'm not sure if it's desirable.

Yep, this is an argument for moving it back to `INFO` level

Reply:
> 2. I've improved the `signAndSend` function (that assigns the `this.#tx` field) to guarantee that it'll emit an error event if some error occurs;

Oh okay, great!

> Maybe we could remove this warn.

Oh remove it, not just change it to an `INFO` log? Maybe if we `INFO` log when sending and `INFO` log when listening we could remove it here.